Add Date and time guidance (#3503)
grommet/hpe-design-system: aries-site/src/pages/foundation/date-and-time.mdx

import { BestPracticeGroup, Example } from '../../layouts';
import {
  AbbreviatedDateExample,
  DateTimeNotificationExample,
  RelativeTimeExample,
  TimezoneExample,
  ZeroPaddingExample,
} from '../../examples';
import { Text } from 'grommet';

Date and time can appear in a wide variety of contexts:

- Data tables
- Inline with paragraph text
- Name value lists
- Notifications
- Log files

Depending on the context, date and time may be presented as:

| Format                          | Purpose                                                                         | Examples                      | Notes                                                                              |
| ------------------------------- | ------------------------------------------------------------------------------- | ----------------------------- | ---------------------------------------------------------------------------------- |

> | [Absolute time](#absolute-time) | To provide the precise date and/or time of an event.                            | September 15, 2023; 3/27/2023 | The granularity and formatting depends on the context. See absolute time guidance. |

Reviewer: ```suggestion
| [Absolute time](#absolute-time) | To provide the precise date and/or time of an event.                            | September 15, 2023; 3/27/2023 | The granularity and formatting depends on the context. See [absolute time guidance](#absolute-time). |
```

Author: Done.

> | [Absolute time](#absolute-time) | To provide the precise date and/or time of an event.                            | September 15, 2023; 3/27/2023 | The granularity and formatting depends on the context. See absolute time guidance. |

Reviewer: Should we be using the leading zero here? Also, should we be using Intl.DateTimeFormat() to present the dates?

```suggestion
| [Absolute time](#absolute-time) | To provide the precise date and/or time of an event.                            | September 15, 2023; 03/27/2023 | The granularity and formatting depends on the context. See absolute time guidance. |
```

Author: We can update to use `Intl.DateTimeFormat()`. But I think we are only recommending using leading zero if it's in a comparison, like in a column.

Author: Done.

Author: Commented somewhere else on this. but I believe we only want to use the zero padded in a column setting.

Author: Updated to use Date method.

Reviewer 2: This really depends on the context here. Considering this is standalone in the table, I'd actually expect it to not have a leading zero. Might be easier to just put like 10/27/2023 instead to avoid confusion in this section.

Author: Done.

export const date = new Date('2023-08-12');

| Formats                        | Example                                                                                                           | When to use                                                                                                                                                                                                                                                                            | Code snippet (where date is a Javascript Date object)                                                                            |
| ------------------------------ | ----------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- |

> | Long written format            | {Intl.DateTimeFormat('en-US', { weekday: 'short', month: 'long', day: 'numeric', year: 'numeric' }).format(date)} | When space is available and displaying the day of week is beneficial to the user.                                                                                                                                                                                                      | <Text>{`Intl.DateTimeFormat('en-US', { weekday: 'short',  day: 'numeric',month: 'long', year: 'numeric' }).format(date)`}</Text> |

Author: I'll give it a try to format it like that. In general, do you think having the code snippet is valuable?

> | Long written format            | {Intl.DateTimeFormat('en-US', { weekday: 'short', month: 'long', day: 'numeric', year: 'numeric' }).format(date)} | When space is available and displaying the day of week is beneficial to the user.                                                                                                                                                                                                      | <Text>{`Intl.DateTimeFormat('en-US', { weekday: 'short',  day: 'numeric',month: 'long', year: 'numeric' }).format(date)`}</Text> |

Reviewer: Wondering if instead of hard coding to `en-US`, we should do 
```
const locale = navigator.language;

....

`Intl.DateTimeFormat(locale, { weekday: 'short'`Intl.DateTimeFormat('en-US', { weekday: 'short' ...`
```

OR, I think using `undefined` uses the browser's default locale.

```
`Intl.DateTimeFormat(undefined, { weekday: 'short'`Intl.DateTimeFormat('en-US', { weekday: 'short' ...`
```

> | Long written format            | {Intl.DateTimeFormat('en-US', { weekday: 'short', month: 'long', day: 'numeric', year: 'numeric' }).format(date)} | When space is available and displaying the day of week is beneficial to the user.                                                                                                                                                                                                      | <Text>{`Intl.DateTimeFormat('en-US', { weekday: 'short',  day: 'numeric',month: 'long', year: 'numeric' }).format(date)`}</Text> |

Author: Done.

> | Long written format            | {Intl.DateTimeFormat('en-US', { weekday: 'short', month: 'long', day: 'numeric', year: 'numeric' }).format(date)} | When space is available and displaying the day of week is beneficial to the user.                                                                                                                                                                                                      | <Text>{`Intl.DateTimeFormat('en-US', { weekday: 'short',  day: 'numeric',month: 'long', year: 'numeric' }).format(date)`}</Text> |

Author: Done.

| Medium written format          | {Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(date)}                                                | When space is available.                                                                                                                                                                                                                                                               | <Text>{`Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(date)`}</Text>                                                |
| Short written format           | {Intl.DateTimeFormat('en-US', { dateStyle: 'medium' }).format(date)}                                              | When space is limited.                                                                                                                                                                                                                                                                 | <Text>{`Intl.DateTimeFormat('en-US', { dateStyle: 'medium' }).format(date)`}</Text>                                              |

> | Numeric locale-specific format | {Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)}                | When the user needs to compare across dates, such as in a column or list. Otherwise, use written format. Numeric format can differ depending on the region, so even if the date is translated by the UI based on the user's locale, numeric dates can still cause confusion for users. | <Text>{`Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)`}</Text>                |

Reviewer: ```suggestion
| Numeric locale-specific format | {Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)}                | When the user needs to compare across dates, such as in a column or list. Otherwise, use written format. <br /> Numeric format can differ depending on the region, so even if the date is translated by the UI based on the user's locale, numeric dates can still cause confusion for users. | <Text>{`Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)`}</Text>                |
```

I'm unclear as to what the third sentence "Numeric format can differ depending on the region, so even if the date is translated by the UI based on the user's locale, numeric dates can still cause confusion for users." is attempting to communicate.

Author: I think the intent was to say that written format is a more reliable way to convey a date. The benefits of numeric format are really just for ease of quick comparison. I've removed the third sentence for now, just ended as "Otherwise, use written format."

Reviewer 2: @halocline poor wording on my end. Wanted to convey that numeric formatting is untrustworthy and confusing for users. Say that its 8/12 in US formatting but as a user in a different region you'd expect 15/8 instead. If the system shows 12/8 you might be wondering whether the date format has translated to your locale or if it's using a different region's formatting. Overall written format is more reliable as Taylor mentioned.

| ------------------------------- | ------------------------------------------------------------------------------- | ----------------------------- | ---------------------------------------------------------------------------------- |
| [Absolute time](#absolute-time) | To provide the precise date and/or time of an event.                            | September 15, 2023; 3/27/2023 | The granularity and formatting depends on the context. See absolute time guidance. |
| [Relative time](#relative-time) | To display how long ago something occurred or in how long something will occur. | in 5 days, 3 days ago         | --                                                                                 |

Timestamps should always be stored in UTC, the universal standard. These UTC timestamps will then get converted to local time, if necessary, by the UI for display.

## Absolute time

Absolute time stamps display the precise date and time an action has occurred or will occur.

May be presented as:

- Date
- Date and time

> - Including seconds (and fractional seconds)
> - Including timezone

Reviewer: Are these sub bullets of "Date and time?"

```suggestion
  - Including seconds (and fractional seconds)
  - Including timezone
```

export const date = new Date('2023-08-12');

| Formats                        | Example                                                                                                           | When to use                                                                                                                                                                                                                                                                            | Code snippet (where date is a Javascript Date object)                                                                        |
| ------------------------------ | ----------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------- |

> | Long written format            | {Intl.DateTimeFormat('en-US', { weekday: 'short', month: 'long', day: 'numeric', year: 'numeric' }).format(date)} | When space is available and displaying the day of week is beneficial to the user.                                                                                                                                                                                                      | <Box>`Intl.DateTimeFormat('en-US', { weekday: 'short',  day: 'numeric',month: 'long', year: 'numeric' }).format(date)`</Box> |

Reviewer: ```suggestion
| Full written format            | {Intl.DateTimeFormat('en-US', { weekday: 'short', month: 'long', day: 'numeric', year: 'numeric' }).format(date)} | When space is available and displaying the day of week is beneficial to the user.                                                                                                                                                                                                      | <Box>`Intl.DateTimeFormat('en-US', { dateStyle: 'full' }).format(date)`</Box> |
```

Author: Good suggestions, updating.

> | Medium written format          | {Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(date)}                                                | When space is available.                                                                                                                                                                                                                                                               | <Box>`Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(date)`</Box>                                                |

Reviewer: Align labels to browser API ?

```suggestion
| Long written format          | {Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(date)}                                                | When space is available.                                                                                                                                                                                                                                                               | <Box>`Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(date)`</Box>                                                |
```

> | Short written format           | {Intl.DateTimeFormat('en-US', { dateStyle: 'medium' }).format(date)}                                              | When space is limited.                                                                                                                                                                                                                                                                 | <Box>`Intl.DateTimeFormat('en-US', { dateStyle: 'medium' }).format(date)`</Box>                                              |

Reviewer: Align labels to browser API ?

```suggestion
| Medium written format           | {Intl.DateTimeFormat('en-US', { dateStyle: 'medium' }).format(date)}                                              | When space is limited.                                                                                                                                                                                                                                                                 | <Box>`Intl.DateTimeFormat('en-US', { dateStyle: 'medium' }).format(date)`</Box>                                              |
```

| Numeric locale-specific format | {Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)}                | When the user needs to compare across dates, such as in a column or list. Otherwise, use written format. Numeric format can differ depending on the region, so even if the date is translated by the UI based on the user's locale, numeric dates can still cause confusion for users. | <Box>`Intl.DateTimeFormat('en-US', { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)`</Box>                |

> | ISO 8601 format                | {date.toISOString()}                                                                                              | Log files; trigger times.                                                                                                                                                                                                                                                              | <Box>`date.toISOString()`</Box>                                                                                              |

Reviewer: Can we initialize date with a time so that this example presents something with a more realistic timestamp?

Author: Yes will update.

> | ISO 8601 format                | {date.toISOString()}                                                                                              | Log files; trigger times.                                                                                                                                                                                                                                                              | <Box>`date.toISOString()`</Box>                                                                                              |

Reviewer: Why Box and not Text? Also, can we prevent the timestamp from wrapping? I don't think we'd want timestamps to wrap in a table.

```suggestion
| ISO 8601 format                | {date.toISOString()}                                                                                              | Log files; trigger times.                                                                                                                                                                                                                                                              | <Box>`date.toISOString()`</Box>                                                                                              |
```

Author: Switched to Text and fixed wrapping. Had to use an inline style since `whiteSpace` isn't a prop supported by Text.

Author: Done.

Other than log files, date and time should be presented according to the [user's locale](#locale-formatting) and [timezone](#timezones).

For log files, there are two options:

1. Always output timestamps in UTC, indicating UTC in the output.
1. Allow the user to specify a default timezone and always output timestamps in that timezone, specifying the timezone in the output.

### Locale formatting

> Date and time are formatted differently depending on the language and region. Some aspects that are locale-dependent include date and time ordering, separators, and 12-hour vs 24-hour clocks.

Reviewer: ```suggestion
Applying the user experience principles of "personalization" and "don't make me think," in most cases dates and times should be presented based on the user's locale as specified in their browser and system preferences.

Date and time are formatted differently depending on the language and region. Some aspects that are locale-dependent include date and time ordering, separators, and 12-hour vs 24-hour clocks.
```

There are various [Javascript Date methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) that can be leveraged to convert from UTC to the user's locale.

Whenever possible, present date and time according to the user's locale.

export const localeDate = new Date('2023-09-15T12:00:00.000Z');

> export const localeFormat = {
>   year: 'numeric',
>   month: 'long',
>   day: 'numeric',
>   weekday: 'long',
> };

Reviewer: Is this the equivalent of 

```suggestion
export const localeFormat = {
  dateStyle: 'full'
};
```

> | Locale | Date                                                            |

Reviewer: Would it be helpful to add a DateTime column as well?

Author: Assuming this means the code snippet? Yes I can add.

Author: Done.

Reviewer: I was actually referring to presenting a localized date + time. The content above speaks to localizing date and time, so was thinking that in addition to the localized date column, we'd do a localized date and time column which would likely present concepts like 24 hour clock and timezone.

Author: Thanks for clarifying, will update so time is also include

| Numeric locale-specific format | {Intl.DateTimeFormat(undefined, { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)} | When the user needs to compare across dates, such as in a column or list. Otherwise, use written format. | <Text>`Intl.DateTimeFormat(undefined, { month: '2-digit', day: '2-digit', year: 'numeric' }).format(date)`</Text> |
| ISO 8601 format                | <Text style={{whiteSpace: 'nowrap'}}>{date.toISOString()}</Text>                                     | Log files; trigger times.                                                                                | <Text>`date.toISOString()`</Text>                                                                                 |

Other than log files, date and time should be presented according to the [user's locale](#locale-formatting) and [timezone](#timezones).

For log files, there are two options:

1. Always output timestamps in UTC, indicating UTC in the output.

> 1. Allow the user to specify a default timezone and always output timestamps in that timezone, specifying the timezone in the output.

Reviewer 2: I'd be careful to include this option for log files. The issue with using a timezone for something that occurred in the past is that it becomes confusing whenever a timezone is changed or if Daylight Savings has occurred. Using UTC in cases such as log files is a much more consistent throughout.

Author: I can remove this for now if it's not consistent with your research. I had pulled this from some of Edgar's content and was thinking he must've had some background from products he's worked on.

Reviewer 2: From what I have seen I don't agree with that content but can definitely get his feedback as well. I can share with him the deployment link.

Author: I'm fine just removing it for now so we can get this page out. Can always iterate in future.

Reviewer 2: Thinking about this a little bit more and stumbled on this statement:

"- Whenever you are referring to an exact moment in time, persist the time according to a unified standard that is not affected by daylight savings. (GMT and UTC are equivalent with this regard, but it is preferred to use the term UTC. Notice that UTC is also known as Zulu or Z time.)
- If instead you choose to persist a (past) time using a local time value, include the local time offset for this particular time from UTC (this offset may change throughout the year), such that the timestamp can later be interpreted unambiguously."

So yes we can show local time as long as it's offset from UTC (ex. UTC+3). Furthermore, should we clarify in the first bullet point that the timestamp should be displaying UTC+0 specifically? Might also be helpful to include a statement along the lines of "Displaying time zones other than UTC+0 time within log files may present issues if the time zone has changed or if Day Light Savings time has occurred."

Author: This topic feels like it needs more discussion. After connecting with Kenny offline, we'll leave this discussion for future. Removing ambiguous wording about log files though.

Author: Based on discussion, removed the ambiguous wording but the above comment needs a bit more talking through. Let's address separately.

| en-AU  | {Intl.DateTimeFormat('en-AU', localeFormat).format(localeDate)} | <Text>`Intl.DateTimeFormat('en-AU', { dateStyle: 'full' }).format(date)`</Text> |
| de     | {Intl.DateTimeFormat('de', localeFormat).format(localeDate)}    | <Text>`Intl.DateTimeFormat('de', { dateStyle: 'full' }).format(date)`</Text>    |

### Timezones

In addition to formatting a date according to the user's locale, present time in the user's timezone and display the timezone alongside the time stamp for clarity.

In instances when a user's timezone cannot be determined, output the date and time in UTC+0.

<Notification
  status="info"
  message="The same moment in time will be presented differently based on the user's timezone and locale."
  width={{ max: 'large' }}
  margin={{ bottom: 'medium' }}
/>

<BestPracticeGroup>
  <Example
    height="none"
    bestPractice={{
      type: 'do',
      message:
        'Utilize the browser locale to format and update the date and time for users, providing a personalized experience.',
    }}
  >
    <TimezoneExample />
  </Example>
  <Example
    height="none"
    bestPractice={{
      type: 'dont',

>       message: `Don't present a date in a different timezone than that of the user. This leads to confusion, increased cognitive load, and sets users up for failure.`,

Reviewer 2: Originally, I had the same visual examples for the do and don't sections. Received some feedback saying that it was a bit confusing that both examples had the same time for the LA times, leading me to change the don't section into a completely different starting time for LA. I'm indifferent either way but wanted to bring this up.

Author: Gotcha, I can revisit.

Author: Done.

    }}
  >
    <TimezoneExample bestPractice={false} />
  </Example>
</BestPracticeGroup>

### ISO 8601 Formatting

> ISO 8601 is an internationally recognized standard for presenting date and time. It is presented as `YYYY-MM-DDTHH:mm:ss.sssZ`. For example, `2023-02-07T06:57:00.000Z`.

Author: I can clarify in a subsequent sentence. I felt the table and anatomy diagram might be a little overkill

> ISO 8601 is an internationally recognized standard for presenting date and time. It is presented as `YYYY-MM-DDTHH:mm:ss.sssZ`. For example, `2023-02-07T06:57:00.000Z`.

Reviewer 2: Agree with it being overkill, would definitely make a note about those letters though. Might need a sentence on RFC 3339 formatting too but let's connect on that more.

> ISO 8601 is an internationally recognized standard for presenting date and time. It is presented as `YYYY-MM-DDTHH:mm:ss.sssZ`. For example, `2023-02-07T06:57:00.000Z`.

Reviewer: Hmmm... I'm not sure I follow the concern. Doesn't [RFC 3339 point to ISO 8601](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) where T and Z are spec?

> ISO 8601 is an internationally recognized standard for presenting date and time. It is presented as `YYYY-MM-DDTHH:mm:ss.sssZ`. For example, `2023-02-07T06:57:00.000Z`.

Author: I've clarified what `T` and `Z` are.

> ISO 8601 is an internationally recognized standard for presenting date and time. It is presented as `YYYY-MM-DDTHH:mm:ss.sssZ`. For example, `2023-02-07T06:57:00.000Z`.

Reviewer 2: @halocline a reviewer mentioned that they were unfamiliar with what the "T" was since RFC 3339 formatting uses a space instead.

> ISO 8601 is an internationally recognized standard for presenting date and time. It is presented as `YYYY-MM-DDTHH:mm:ss.sssZ`. For example, `2023-02-07T06:57:00.000Z`.

Reviewer: RFC 3339 "allows for a space for readability" but the spec is still based on the TZ of ISO. Seems like we should be sticking to the spec if possible and we reference the ISO spec being followed, so readers have that as a resource for further education. I think we should be good.

| Within the last minute | Just now; a minute ago           | now; a min ago                  |
| 1 day ago              | yesterday                        | yesterday                       |
| < 1 week ago           | x days, hour, minute, second ago | x days, hr, min, sec ago        |
| >= 1 week ago          | [Absolute time](#absolute-time)  | [Absolute time](#absolute-time) |
| In 1 day               | tomorrow                         | tomorrow                        |
| < 1 week from now      | in x days, hour, minute, second  | in x days, hr, min sec          |
| In >= 1 week           | [Absolute time](#absolute-time)  | [Absolute time](#absolute-time) |

<BestPracticeGroup>
  <Example
    height="none"
    bestPractice={{
      type: 'do',
      message:
        'Provide users with only relevant information. In this use case, users do not need to know how many seconds ago an activity was last performed or the exact time activities older than a month were performed.',
    }}
  >
    <RelativeTimeExample />
  </Example>
  <Example
    height="none"
    bestPractice={{
      type: 'dont',

>       message: `Don't include additional information if it isn't needed. This will hurt readability and cause issues with layouts at smaller breakpoints.`,

Reviewer 2: Originally I had this as right aligned but I know we haven't defined that quite yet. Just flagging for future consideration if we want to realign this example.

Author: I've right aligned this based on discussion to add the alignment section.

Zero-padding is not needed when context contains only one time stamp, see [notification example](#notification).

<BestPracticeGroup>
  <Example
    height="none"
    bestPractice={{
      type: 'do',
      message:
        'Zero-pad both date and timestamps for easier comparison across rows.',
    }}
  >
    <ZeroPaddingExample />
  </Example>
  <Example
    height="none"
    bestPractice={{
      type: 'dont',
      message: `Don't omit the leading zero for the date or time. This causes issues with alignment and layout which prevents users from being able to quickly scan and compare the information.`,
    }}
  >
    <ZeroPaddingExample bestPractice={false} />
  </Example>
</BestPracticeGroup>

## Examples

### Notifications

When displaying future events such as upcoming updates, expiration dates, and ongoing processes:

- Use the friendlier, more conversational [relative time]() phrasing, such as “in x days”.
- Provide the date, exact time, and timezone alongside the relative time to remove any ambiguity.

<BestPracticeGroup columns={['auto']}>
  <Example
    height="none"
    bestPractice={{
      type: 'do',
      message:

>         'Accompany the relative time with the exact date, time, and timezone in which the event will occur.',

Reviewer 2: Should the width of the content here expand to the edge of the notification container or should the notification container's width be shortened?

Author: I think that's coming from the theme which causes the notification message to wrap. Let me connect with @halocline on it.

    }}
  >
    <TimezoneExample bestPractice={false} />
  </Example>
</BestPracticeGroup>

### ISO 8601 Formatting

ISO 8601 is an internationally recognized standard for presenting date and time. It is presented as `YYYY-MM-DDTHH:mm:ss.sssZ`.

For example, `2023-02-07T06:57:00.000Z`.

In this format, `T` indiciates the start of the time stamp, and `Z` indiciates the time is in UTC.

Some use cases include:

- Log files
- Trigger times

Even though ISO 8601 is an internationalized standard, it is not the most friendly format for users. In most other cases, date and time formatting should be based on the user's [locale](#locale-formatting) and [timezone](#timezones).

## Relative time

Relative time stamps display:

1. How recently an event has occurred.
1. How far in the future an event will occur.

Relative time stamps remove the need for the user to perform mental calculations by presenting timestamps in written form like "2 days ago".

When displaying relative time:

- Adapt and show the appropriate unit of time. For example, after 60 minutes, display hours.

> - Unless required, displaying the time for anything less than a second and more than a week is generally not needed.

Reviewer 2: Another bullet point here stating that "Zero values should not be included as this will cause issues with alignment. Ex. do not show when something is at 0 minutes."

Author: Done.

  </Example>
</BestPracticeGroup>

## Abbreviating date and time

When space in the UI is limited, units of time, days of the week, and months can be abbreviated.

Some contexts in which abbreviations may be needed are:

- Within cells of a data table
- At smaller breakpoints

| Units of time  | Abbreviation |
| -------------- | ------------ |
| millisecond(s) | ms           |
| second(s)      | sec          |
| minute(s)      | min          |
| hour(s)        | hr           |
| day(s)         | day, days    |
| week(s)        | wk           |
| month(s)       | mo           |
| year(s)        | yr           |

To abbreviate days of the week or months, use the [Javascript Intl.DateTimeFormat method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat). The general convention for English is to use the first 3 letters of the day or month name.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/foundation/date-and-time/AbbreviatedDateExample.js"
  horizontalLayout
  height="none"
>
  <AbbreviatedDateExample />
</Example>

## Alignment

> When presenting dates in a column, right-align [relative times](#relative-time) to bring units of the same magnitude into alignment.

Reviewer 2: Should we mention that by default it should be left aligned? Maybe mention that when using things like ISO 8601 formatting, zero-padded dates, and generally when in comparison mode, it should be left aligned.

Might be helpful to have a similar example to what I had in the figma to show a table that might have both left and right alignment. Purpose of this example was to show the use case "trigger times" and "duration" as well. 

![image](https://github.com/grommet/hpe-design-system/assets/86075038/94599830-6987-478e-9c78-a9cd38598955)

Author: Done.
